robots/cmd/kubevirt-job-copier/main.go

```diff
@@ -0,0 +1,323 @@
+		sourceJobName := createPeriodicJobName(sourceProviderReleaseSemver, sigName)
+
+		if _, exists := allPeriodicJobs[targetJobName]; exists {
+			log.WithField("targetJobName", targetJobName).WithField("sourceJobName", sourceJobName).Info("Target job exists, nothing to do")
+			continue
+		}
+
+		if _, exists := allPeriodicJobs[sourceJobName]; !exists {
+			log.WithField("targetJobName", targetJobName).WithField("sourceJobName", sourceJobName).Warn("Source job does not exist, can't copy job definition!")
+			continue
+		}
+
+		log.WithField("targetJobName", targetJobName).WithField("sourceJobName", sourceJobName).Info("Copying source to target job")
+
+		newJob := config.Periodic{}
+		newJob.Annotations = make(map[string]string)
+		for k, v := range allPeriodicJobs[sourceJobName].Annotations {
+			newJob.Annotations[k] = v
+		}
+		newJob.Cluster = allPeriodicJobs[sourceJobName].Cluster
+		newJob.Cron = advanceCronExpression(allPeriodicJobs[sourceJobName].Cron)
+		newJob.Decorate = allPeriodicJobs[sourceJobName].Decorate
+		newJob.DecorationConfig = allPeriodicJobs[sourceJobName].DecorationConfig.DeepCopy()
+		copy(newJob.ExtraRefs, allPeriodicJobs[sourceJobName].ExtraRefs)
+		newJob.Labels = make(map[string]string)
+		for k, v := range allPeriodicJobs[sourceJobName].Labels {
+			newJob.Labels[k] = v
+		}
+		newJob.MaxConcurrency = allPeriodicJobs[sourceJobName].MaxConcurrency
+		newJob.ReporterConfig = allPeriodicJobs[sourceJobName].ReporterConfig.DeepCopy()
+		newJob.Spec = allPeriodicJobs[sourceJobName].Spec.DeepCopy()
+
+		for index, envVar := range newJob.Spec.Containers[0].Env {
+			if envVar.Name != "TARGET" {
+				continue
+			}
+			newEnvVar := *envVar.DeepCopy()
+			newEnvVar.Value = createTargetValue(targetProviderReleaseSemver, sigName)
+			newJob.Spec.Containers[0].Env[index] = newEnvVar
+			break
+		}
+		newJob.Name = targetJobName
+		jobConfig.Periodics = append(jobConfig.Periodics, newJob)
+
+		updated = true
+	}
+
+	return
+}
+
+func createPresubmitJobName(latestReleaseSemver *querier.SemVer, sigName string) string {
+	return fmt.Sprintf("pull-kubevirt-e2e-k8s-%s.%s-%s", latestReleaseSemver.Major, latestReleaseSemver.Minor, sigName)
+}
+
+func createPeriodicJobName(latestReleaseSemver *querier.SemVer, sigName string) string {
+	return fmt.Sprintf("periodic-kubevirt-e2e-k8s-%s.%s-%s", latestReleaseSemver.Major, latestReleaseSemver.Minor, sigName)
+}
+
+func createTargetValue(latestReleaseSemver *querier.SemVer, sigName string) string {
+	return fmt.Sprintf("k8s-%s.%s-%s", latestReleaseSemver.Major, latestReleaseSemver.Minor, sigName)
+}
+
+// advanceCronExpression advances source cron expression to +1h10m
+// cron expression must have format of i.e. "0 1,9,17 * * *" or it will panic
+func advanceCronExpression (sourceCronExpr string) string {
+	if !cronRegex.MatchString(sourceCronExpr) {
+		logrus.StandardLogger().WithField("cronRegex", cronRegex).WithField("sourceCronExpr", sourceCronExpr).Fatal("cronRegex doesn't match")
```

Comment:
@dhiller Would we just log this mismatch event or do we need to return/panic as well?

Reply:
[logrus.Fatal](https://github.com/sirupsen/logrus/blob/master/entry.go#L335) exits with non zero exit code. Should I rather panic here?

Reply:
@dhiller that will good enough IMHO. Thanks!